ossindex/internal/cache/cache_test.go

```diff
@@ -0,0 +1,122 @@
+//
+// Copyright 2020-present Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package cache has definitions and functions for processing the OSS Index Feed
+package cache
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/shopspring/decimal"
+	"github.com/sonatype-nexus-community/nancy/types"
+	"github.com/stretchr/testify/assert"
+)
+
+var coordinates []types.Coordinate
+
+var purls []string
+
+func TestInsert(t *testing.T) {
+	cache := setupTestsAndCache(t)
+
+	err := cache.Insert(coordinates)
+	assert.Nil(t, err)
+
+	var result DBValue
+	err = cache.getKeyAndHydrate(coordinates[0].Coordinates, &result)
+
+	assert.Equal(t, coordinates[0], result.Coordinates)
+	assert.Nil(t, err)
+
+	tearDown(t, cache)
+}
+
+func TestGetWithRegularTTL(t *testing.T) {
+	cache := setupTestsAndCache(t)
+
+	err := cache.Insert(coordinates)
+	assert.Nil(t, err)
+
+	newPurls, results, err := cache.GetCacheValues(purls)
+
+	assert.Empty(t, newPurls)
+	assert.Equal(t, results, coordinates)
+	assert.Nil(t, err)
+
+	tearDown(t, cache)
+}
+
+func TestGetWithExpiredTTL(t *testing.T) {
+	cache := setupTestsAndCache(t)
+	cache.TTL = time.Now().AddDate(0, 0, -1)
+
+	err := cache.Insert(coordinates)
+	assert.Nil(t, err)
+
+	newPurls, results, err := cache.GetCacheValues(purls)
+
+	assert.Equal(t, purls, newPurls)
+	assert.Empty(t, results)
+	assert.Nil(t, err)
+
+	var result DBValue
+	err = cache.getKeyAndHydrate(purls[0], &result)
+	assert.NotNil(t, err)
+	if _, ok := err.(*os.PathError); ok {
```

Comment:
minor: Not sure I understand why we check for "not" os.PathError here. Is the intent to check for some other specific error type? I would expect to verify a "key not found" or some such error.

Reply:
This check looks for an error (see assert.NotNil() above), and verifies that it's type of `*os.PathError` as that is what is sent back when a key is not found. After `GetCacheValues` is run, if an expired key is run into, it deletes it from Pudge, so it verifies that happens inside the method (a lil granular, but I figured it was worth it)

Reply:
And turns out YOU RIGHT: e9f2708